File: external/webroot/md/codeExample/OCR-API-Bankcard/BankCard.cs
using System;
using System.IO;
using System.Net;
using System.Text;
using System.Web;

namespace com.baidu.ai
{
	public class BankCard
	{
		//银行卡识别
		public static string bankCard()
		{
			string token = "########";
			string strbaser64 = FileUtils.getFileBase64("/work/ai/images/ocr/bankcard.jpeg"); // 图片的base64编码
			string host = "https://aip.baidubce.com/rest/2.0/ocr/v1/bankcard?access_token=" + token;
			Encoding encoding = Encoding.Default;
			HttpWebRequest request = (HttpWebRequest)WebRequest.Create(host);
			request.Method = "post";
			request.ContentType = "application/x-www-form-urlencoded";
			request.KeepAlive = true;
			String str = "image=" + HttpUtility.UrlEncode(strbaser64);
			byte[] buffer = encoding.GetBytes(str);
			request.ContentLength = buffer.Length;
			request.GetRequestStream().Write(buffer, 0, buffer.Length);
			HttpWebResponse response = (HttpWebResponse)request.GetResponse();
			StreamReader reader = new StreamReader(response.GetResponseStream(), Encoding.Default);
			string result = reader.ReadToEnd();
			Console.WriteLine("银行卡识别:");
			Console.WriteLine(result);
			return result;
		}
	}
}

File: external/webroot/md/codeExample/OCR-API-Bankcard/BankCard.py
# coding:utf-8
import urllib, urllib2, sys
import ssl

url = 'https://aip.baidubce.com/rest/2.0/ocr/v1/bankcard?access_token=24.a1ff1c9e242129e0e88e506dfc223454.2592000.1492586050.282335-9395294'
bodys = {}
bodys[''] = "{\"image\":\"【图片base64编码】\"}"
post_data = bodys['']
request = urllib2.Request(url, post_data)
request.add_header('Content-Type', 'application/x-www-form-urlencoded')
ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE
response = urllib2.urlopen(request, context=ctx)
content = response.read()
if (content):
    print(content)
